routes.jsx

```diff
@@ -15,21 +16,43 @@ import Navbar from './components/navbar/navbar.jsx';
 import Footer from './components/footer/footer.jsx';
 
 const App = React.createClass({
+  getInitialState() {
+    return {
+      suppressSplashScreen: (localstorage.getItem(`suppressSplashScreen`) === `true`)
+    };
+  },
+  getDefaultProps() {
+    return {
+      dismissTimeout: 3000
+    };
+  },
   componentDidMount() {
-    setTimeout(function() {
-      this.splash.classList.add(`dismissed`);
-      document.querySelector(`#app`).classList.add(`splash-dismissed`);
-    }, 3000);
+    if (!this.state.suppressSplashScreen) {
+      setTimeout(this.dismissSplash, this.props.dismissTimeout);
+    }
+  },
+  dismissSplash() {
+    this.splash.classList.add(`dismissed`);
```

Comment:
is there a way for us to tie a `setState({ suppressSplashScreen: true })` to when this CSS transition finishes?